cloudpathlib/azure/azblobclient.py

```diff
@@ -200,30 +210,30 @@ def _list_dir(
         if prefix and not prefix.endswith("/"):
             prefix += "/"
 
-        yielded_dirs = set()
-
-        # NOTE: Not recursive may be slower than necessary since it just filters
-        #   the recursive implementation
-        for o in container_client.list_blobs(name_starts_with=prefix):
-            # get directory from this path
-            for parent in PurePosixPath(o.name[len(prefix) :]).parents:
-                # if we haven't surfaced this directory already
-                if parent not in yielded_dirs and str(parent) != ".":
-                    # skip if not recursive and this is beyond our depth
-                    if not recursive and "/" in str(parent):
-                        continue
-
-                    yield (
-                        self.CloudPath(f"az://{cloud_path.container}/{prefix}{parent}"),
-                        True,  # is a directory
-                    )
-                    yielded_dirs.add(parent)
-
-            # skip file if not recursive and this is beyond our depth
-            if not recursive and "/" in o.name[len(prefix) :]:
-                continue
-
-            yield (self.CloudPath(f"az://{cloud_path.container}/{o.name}"), False)  # is a file
+        if not recursive:
+            blobs = container_client.walk_blobs(name_starts_with=prefix)
```

Comment:
Does this introduce an alternative minimum sdk version to what we have?
https://github.com/drivendataorg/cloudpathlib/blob/master/pyproject.toml#L36

Reply:
No, it does not:
https://github.com/Azure/azure-sdk-for-python/blob/0ab186f98cf2c4c83c1b4260406027d9c4f8cc06/sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py#L652